src/core/music_score_events.h

```diff
@@ -0,0 +1,668 @@
+#pragma once
+
+#include <vector>
+#include <algorithm>
+#include <unordered_map>
+#include <memory>
+
+template <typename EventType> class MusicScoreEventBrowser;
+
+
+typedef int32_t BarNumber;
+typedef int32_t BeatNumber;
+
+/**
+ * Marks a specific beat of a specific bar of music.
+ */
+struct BeatAndBar {
```

Comment:
Perhaps we might be able to make this simpler by using std::pair or std::tuple.  They have comparisons built in (operator==, <), etc.  It could be done:

  using BeatAndBar = std::pair<BarNumber,BeatNumber>;
  
  const BarNumber& GetBar(const BeatAndBar& b) { return b.first; }
  BarNumber& GetBar(BeatAndBar& b) { return b.first; }
  const BeatNumber& GetBeat(const BeatAndBar& b) { return b.second; }
  BeatNumber& GetBeat(BeatAndBar& b) { return b.second; }